tests/test_handle_error.py

```diff
@@ -0,0 +1,86 @@
+import unittest
+from unittest.mock import Mock
+import asyncio
+import argparse
+import aiohttp
+import shutil
+import os
+from snare import HttpRequestHandler
+
+
+class AsyncMock(Mock):  # custom function defined to mock asyncio coroutines
+
+    def __call__(self, *args, **kwargs):
+        sup = super(AsyncMock, self)
+
+        async def coro():
+            return sup.__call__(*args, **kwargs)
+        return coro()
+
+    def __await__(self):
+        return self().__await__()
+
+
+class TestHandleError(unittest.TestCase):
+    def setUp(self):
+        self.meta = {}
+        run_args = argparse.ArgumentParser()
+        run_args.add_argument("--tanner")
+        run_args.add_argument("--page-dir")
+        if not os.path.exists("/opt/snare/pages/test"):
+            os.makedirs("/opt/snare/pages/test")
+        self.args = run_args.parse_args(['--page-dir', 'test'])
+        self.loop = asyncio.new_event_loop()
+        self.status = 500
+        self.message = "test"
+        self.payload = "test"
+        self.exc = "[Errno 0] test"
+        self.headers = "test"
+        self.reason = "test"
+        self.data = dict(
+            method='GET',
+            path='/',
+            headers="test_headers",
+            uuid="test_uuid",
+            peer="test_peer",
+            status="test_status",
+            error=self.exc
+        )
+        aiohttp.server.ServerHttpProtocol.handle_error = Mock()
+
+    def test_create_error_data(self):
+        self.handler = HttpRequestHandler(self.meta, self.args)
```

Comment:
you have similar setup for handler in all functions - maybe it's better to move it into setup?